Add plugin: Mindmap

OneCalmCloud opened the pull request:
## Repo URL

Link to my plugin: https://github.com/OneCalmCloud/obsidian-mindmap

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [x]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

OneCalmCloud commented:
/skip
These styles assigned via JavaScript or HTML are used for user interaction with the mindmap interface. They do not need to be adaptable based on themes and snippets.

joethei commented:
["minAppVersion": "0.9.12",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/manifest.json#L5)
You are using newer API's, set this to the latest public build number.

["description": "An Obsidian plugin for creating Mindmaps.",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/manifest.json#L6)
Avoid including sentences like "This is a plugin for Obsidian that does ..." in your description, it should be self-evident that this is a plugin for Obsidian.
We have published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.

[this.app.workspace.on("active-leaf-change", (leaf) => {](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/main.tsx#L213)
Events need to be registered, so they can be properly unloaded when the plugin is disabled. <https://docs.obsidian.md/Plugins/Events>

[this.containerEl.createEl("h3", {](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/main.tsx#L272-L274)
[Don't include a header with the plugin name in the settings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)


[this.containerEl.createEl("h4", {](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/main.tsx#L315-L317), [this.containerEl.createEl("h4", {](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/main.tsx#L363-L365), [this.containerEl.createEl("h4", {](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/main.tsx#L411-L413)
For section headings in settings use:
```ts
new Setting(containerEl).setName('name here').setHeading();
```

["Mindmap Setting": "Mindmap Setting",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L5), ["Interface Setting": "Interface Setting",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L11), ["Edit Settings": "Edit Settings",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L20), ["View Settings": "View Settings",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L29)
[Avoid "settings" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)


["Export Map Clarity": "Export Map Clarity",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L9), ["Show Center Button": "Show Center Button",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L12), ["Show Fit-to-Screen Button": "Show Fit-to-Screen Button",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L14), ["Show Download Button": "Show Download Button",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L16), ["Show Undo/Redo Buttons": "Show Undo/Redo Buttons",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L18), ["Add Node Button": "Add Node Button",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L21), ["Drag Nodes": "Drag Nodes",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L23), ["Context Menu": "Context Menu",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L27), ["Sharpen Corners": "Sharpen Corners",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L30), ["Line Width": "Line Width",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L32-L33), ["Vertical Spacing": "Vertical Spacing",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L35), ["Primary Node":"Primary Node"](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L38)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

Why are there two different system for i18n setup in your plugin?

OneCalmCloud commented:
> ["minAppVersion": "0.9.12",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/manifest.json#L5) You are using newer API's, set this to the latest public build number.
> 
> ["description": "An Obsidian plugin for creating Mindmaps.",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/manifest.json#L6) Avoid including sentences like "This is a plugin for Obsidian that does ..." in your description, it should be self-evident that this is a plugin for Obsidian. We have published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.
> 
> [this.app.workspace.on("active-leaf-change", (leaf) => {](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/main.tsx#L213) Events need to be registered, so they can be properly unloaded when the plugin is disabled. https://docs.obsidian.md/Plugins/Events
> 
> [this.containerEl.createEl("h3", {](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/main.tsx#L272-L274) [Don't include a header with the plugin name in the settings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)
> 
> [this.containerEl.createEl("h4", {](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/main.tsx#L315-L317), [this.containerEl.createEl("h4", {](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/main.tsx#L363-L365), [this.containerEl.createEl("h4", {](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/main.tsx#L411-L413) For section headings in settings use:
> 
> ```ts
> new Setting(containerEl).setName('name here').setHeading();
> ```
> 
> ["Mindmap Setting": "Mindmap Setting",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L5), ["Interface Setting": "Interface Setting",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L11), ["Edit Settings": "Edit Settings",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L20), ["View Settings": "View Settings",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L29) [Avoid "settings" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)
> 
> ["Export Map Clarity": "Export Map Clarity",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L9), ["Show Center Button": "Show Center Button",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L12), ["Show Fit-to-Screen Button": "Show Fit-to-Screen Button",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L14), ["Show Download Button": "Show Download Button",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L16), ["Show Undo/Redo Buttons": "Show Undo/Redo Buttons",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L18), ["Add Node Button": "Add Node Button",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L21), ["Drag Nodes": "Drag Nodes",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L23), ["Context Menu": "Context Menu",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L27), ["Sharpen Corners": "Sharpen Corners",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L30), ["Line Width": "Line Width",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L32-L33), ["Vertical Spacing": "Vertical Spacing",](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L35), ["Primary Node":"Primary Node"](https://github.com/OneCalmCloud/obsidian-mindmap/blob/1689d5dd91c040cda2ba05c568aa7b964999c627/src/lang/locale/en.ts#L38) [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)
> 
> Why are there two different system for i18n setup in your plugin?

I have confirmed the correct public build number and updated the 'minAppVersion' to the corresponding public build version.
["minAppVersion": "0.12.17"](https://github.com/OneCalmCloud/obsidian-mindmap/blob/0.3.2/manifest.json#L5)

I have updated the plugin description.
["description": "Create notes with Mindmaps."](https://github.com/OneCalmCloud/obsidian-mindmap/blob/0.3.2/manifest.json#L6)

I have added event registration to ensure proper unloading of event listeners when the plugin is disabled.
https://github.com/OneCalmCloud/obsidian-mindmap/blob/684b21ab3f7b52828f6ad6a6b70c827c0a3e482f/src/main.tsx#L216C1-L242C7

I have formatted section headings in settings correctly.
https://github.com/OneCalmCloud/obsidian-mindmap/blob/684b21ab3f7b52828f6ad6a6b70c827c0a3e482f/src/main.tsx#L318
https://github.com/OneCalmCloud/obsidian-mindmap/blob/684b21ab3f7b52828f6ad6a6b70c827c0a3e482f/src/main.tsx#L364
https://github.com/OneCalmCloud/obsidian-mindmap/blob/684b21ab3f7b52828f6ad6a6b70c827c0a3e482f/src/main.tsx#L410

I have simplified the i18n setup, adopting a unified system.
[import i18n from './i18n'](https://github.com/OneCalmCloud/obsidian-mindmap/blob/0.3.2/src/main.tsx#L7)
